pkg/apis/serving/k8s_validation.go

```diff
@@ -450,9 +461,15 @@ func validateVolumeMounts(mounts []corev1.VolumeMount, volumes sets.String) *api
 		if !vm.ReadOnly {
 			errs = errs.Also(apis.ErrMissingField("readOnly").ViaIndex(i))
 		}
-
 	}
+	if IsInSingleContainer(ctx) {
+		errs = errs.Also(getVolumeMountMissingErrors(volumes, seenName))
+	}
```

Comment:
So it's a bit odd that for multi-container we error one place and for single-container we error in another.

I think the best place to have a canonical place to error would be on the volume itself.

e.g. gather up all of the mounted volumes from all containers (this works regardless of single or multi).  Then in validateVolume we should treat a volume as having an error if it is never mounted.

Reply:
Ok I can do that makes sense.